### ISSUE TITLE
Automatically Archive Projects that are 100% mapped and validated

### ISSUE THREAD

**Issue by SColchester** [hotosm/tasking-manager#6365]
**Problem**
Project creators are instructed to archive projects once they are 100% mapped and validated but they often don't archive their completed projects. This clutters the Tasking Manager with 'Published' projects which cannot be contributed to. Tasking Manager admins sink a lot of time into manually archiving these projects and following up with the project creators. 

**Solution**
Once a project reaches 100% mapped and validated, automatically trigger an immediate status change for that project to shift from 'Published' -> 'Archived' - ideally with a message to the project creator to notify them. 

**Additional context**
There is really no reason to keep a project which is 100% mapped and validated active as a 'Published' project, once projects are archived they can still be found and searched for by the public.

There is a large backlog of 100% mapped and validated projects which are not archived. If this change could also work to change the status of those (as well as newer projects that reach the 100% mark) this would be very valuable and a big time-saver.

**Comment by ramyaragupathy** [comment]
@RAytoun's input: message project creator to do a third pass check for data quality before automatically archiving

**Comment by SColchester** [comment]
Added this issue to [Tasking Manager Project Gardening 2024 Wiki](https://wiki.openstreetmap.org/wiki/Humanitarian_OSM_Team/Working_groups/Data_Quality_Control_and_Assurance/Tasking_Manager_Project_Gardening_2024) mentioning that if this issue is solved it will contribte to making the manual Gardening process redundant in the future - which would be great!

**Comment by anthaas** [comment]
Hi, this is really a great idea. It came to my attention that even if some tasks are marked as bad imagery, it is not considered in the overall completeness. So basically I can end up with 100% mapped and 100% validated but lots of tasks can be marked as bad imagery. In my experience usually I had to undo the bad imagery state and request to map those tiles again.
Would be great to consider a certain time frame between finishing the project and automatic archiving. Because if the last mapped task is marked and it triggers the archiving process immediately, I won’t be able to change the state of the bad imagery tiles.
Hope it makes sense.

**Comment by SColchester** [comment]
Yeah totally @anthaas! I think that a way around this would be for it to only target projects where the total number of tasks = total number of validated tasks. So it's not looking for total mapped = total validated, because that would always archive projects that were validated apart from the bad imagery tasks.
